src/java.base/share/classes/java/util/Properties.java

```diff
@@ -725,7 +725,7 @@ private String saveConvert(String theString,
                     outBuffer.append('\\'); outBuffer.append(aChar);
                     break;
                 default:
-                    if (((aChar < 0x0020) || (aChar > 0x007e)) & escapeUnicode ) {
+                    if (((aChar < 0x0020) || (aChar > 0x007e)) && escapeUnicode ) {
```

Comment:
As soon as `escapeUnicode` is already calculated you could flip `if` expression to prevent range check of `aChar` when it's not necessary